test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/DefaultBindingMetadataProviderTest.cs

```diff
@@ -147,5 +147,350 @@ public void GetBindingDetails_FindsBindingSource_IfNullFallsBack()
+
+            // Assert
+            Assert.False(context.BindingMetadata.IsBindingAllowed);
+            Assert.False(context.BindingMetadata.IsBindingRequired);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GetBindingDetails_BindingBehaviorLeftAlone_ForTypeMetadata(bool initialValue)
+        {
+            // Arrange
+            var attributes = new object[]
+            {
+                new BindingBehaviorAttribute(BindingBehavior.Required),
+            };
+
+            var context = new BindingMetadataProviderContext(
+                ModelMetadataIdentity.ForType(typeof(string)),
+                new ModelAttributes(attributes));
+
+            // These values shouldn't be changed since this is a Type-Metadata
+            context.BindingMetadata.IsBindingAllowed = initialValue;
+            context.BindingMetadata.IsBindingRequired = initialValue;
+
+            var provider = new DefaultBindingMetadataProvider();
+
+            // Act
+            provider.GetBindingMetadata(context);
+
+            // Assert
+            Assert.Equal(initialValue, context.BindingMetadata.IsBindingAllowed);
+            Assert.Equal(initialValue, context.BindingMetadata.IsBindingRequired);
+        }
+
+        // Unlike most model metadata settings, BindingBehavior can be specified on the *container type*
+        // but not on the property type.
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GetBindingDetails_BindingBehaviorLeftAlone_ForAttributeOnPropertyType(bool initialValue)
+        {
+            // Arrange
+            var typeAttributes = new object[]
+            {
+                new BindingBehaviorAttribute(BindingBehavior.Required),
+            };
+
+            var context = new BindingMetadataProviderContext(
+                ModelMetadataIdentity.ForProperty(typeof(int), "Length", typeof(string)),
+                new ModelAttributes(propertyAttributes: new object[0], typeAttributes: typeAttributes));
+
+            // These values shouldn't be changed since this is a Type-Metadata
+            context.BindingMetadata.IsBindingAllowed = initialValue;
+            context.BindingMetadata.IsBindingRequired = initialValue;
+
+            var provider = new DefaultBindingMetadataProvider();
+
+            // Act
+            provider.GetBindingMetadata(context);
+
+            // Assert
+            Assert.Equal(initialValue, context.BindingMetadata.IsBindingAllowed);
+            Assert.Equal(initialValue, context.BindingMetadata.IsBindingRequired);
+        }
+
+        [BindNever]
+        private class BindNeverOnClass
+        {
+            public string Property { get; set; }
+        }
+
+        private class InheritedBindNeverOnClass : BindNeverOnClass
+        {
+        }
+
+        [BindRequired]
+        private class BindRequiredOnClass
+        {
+            public string Property { get; set; }
+        }
```

Comment:
aren't testing precedence of multiple attributes in class hierarchy. would be good to have a test or two using a class like
``` c#
[BindNever]
class BindNeverOnClassTotallyWins : BindRequiredOnClass
{
}
```